packages/platform-android/native_modules.gradle

```diff
@@ -183,21 +183,27 @@ class ReactNativeModules {
     if (this.reactNativeModules != null) return this.reactNativeModules
 
     ArrayList<HashMap<String, String>> reactNativeModules = new ArrayList<HashMap<String, String>>()
-    def npx = Os.isFamily(Os.FAMILY_WINDOWS) ? "npx.cmd" : "npx"
-    def command = "${npx} --quiet --no-install react-native config"
+
     /**
-     * Running npx from the directory of the JS app which holds this script in its node_modules.
-     * We do so, because Gradle may be ran with a different directory as CWD, that's outside of JS project
-     * (e.g. when running with -p flag), in which case npx wouldn't resolve correct `react-native` binary.
+     * Resolve the CLI location from Gradle file
+     *
+     * @todo: Sometimes Gradle can be called outside of the JavaScript hierarchy (-p flag) which
+     * will fail to resolve the script and the dependencies. We should resolve this soon.
+     *
+     * @todo: `fastlane` has been reported to not work too.
      */
-    def dir = new File(this.jsAppDir)
-    def reactNativeConfigOutput = this.getCommandOutput(command, /* dir */) // Temporary disable changing dir, as it introduces regressions
+    def cliResolveScript = "console.log(require('@react-native-community/cli').bin);"
+    def cliPath = this.getCommandOutput("node -e ${cliResolveScript}")
```

Comment:
Can you make sure `node -e` works on Windows?

Reply:
Well, `bin` comes from the Node.js `require.resolve` so it's coming from the API layer. The only question is whether `node` can be accessed on Windows this way or not

Reply:
I have confirmed on a Virtual Machine that running `node -e script` works in both PowerShell and CMD. 

I think we're good to go and ship it as "next"

Reply:
Why not shippining it to current 3.x?